File: testcases_T3/solution.py

import sys
from collections import deque

def solve():
    # Solusi untuk masalah Antrian Puskesmas Pak Dokter.

    try:
        n_str = sys.stdin.readline()
        if not n_str: return
        n = int(n_str)

        emergency_q = deque()
        regular_q = deque()
        registered_numbers = set()
        MAX_PATIENTS = 30

        for _ in range(n):
            line = sys.stdin.readline().strip().split()
            if not line: continue
            command = line[0]

            if command == "DAFTAR":
                number, name, type = int(line[1]), line[2], line[3]
                if number in registered_numbers:
                    print("NOMOR SUDAH ADA")
                elif len(emergency_q) + len(regular_q) >= MAX_PATIENTS:
                    print("ANTRIAN PENUH")
                else:
                    registered_numbers.add(number)
                    patient_data = (number, name)
                    if type == 'E':
                        emergency_q.append(patient_data)
                    else:
                        regular_q.append(patient_data)
                    print("TERDAFTAR")

            elif command == "PANGGIL":
                if emergency_q:
                    number, name = emergency_q.popleft()
                    print(f"PANGGIL {number} {name}")
                elif regular_q:
                    number, name = regular_q.popleft()
                    print(f"PANGGIL {number} {name}")
                else:
                    print("ANTRIAN KOSONG")

            elif command == "LIHAT":
                if not emergency_q and not regular_q:
                    print("ANTRIAN KOSONG")
                else:
                    e_list = " ".join([f"{p[0]}-{p[1]}" for p in emergency_q])
                    r_list = " ".join([f"{p[0]}-{p[1]}" for p in regular_q])
                    print(f"EMERGENCY: {e_list if e_list else 'kosong'}")
                    print(f"REGULAR: {r_list if r_list else 'kosong'}")

            elif command == "CARI":
                number_to_find = int(line[1])
                found = False
                
                pos = 1
                for number, name in emergency_q:
                    if number == number_to_find:
                        print(f"POSISI {pos} EMERGENCY")
                        found = True
                        break
                    pos += 1
                
                if not found:
                    pos = 1
                    for number, name in regular_q:
                        if number == number_to_find:
                            print(f"POSISI {pos} REGULAR")
                            found = True
                            break
                        pos += 1
                
                if not found:
                    print("TIDAK DITEMUKAN")

    except (IOError, ValueError, IndexError):
        return

if __name__ == "__main__":
    solve()

File: testcases_T3/solution.cpp

#include <iostream>
#include <string>
#include <vector>
#include <queue>
#include <set>
#include <sstream>

struct Patient {
    int number;
    std::string name;
};

void print_queue(std::queue<Patient> q) {
    if (q.empty()) {
        std::cout << "kosong";
        return;
    }
    bool first = true;
    while (!q.empty()) {
        if (!first) {
            std::cout << " ";
        }
        Patient p = q.front();
        q.pop();
        std::cout << p.number << "-" << p.name;
        first = false;
    }
}

void setup_io() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
}

int main() {
    setup_io();

    int n;
    std::cin >> n;

    std::queue<Patient> emergency_q;
    std::queue<Patient> regular_q;
    std::set<int> registered_numbers;
    const int MAX_PATIENTS = 30;

    std::string command;
    while (n-- > 0) {
        std::cin >> command;

        if (command == "DAFTAR") {
            int number;
            std::string name, type;
            std::cin >> number >> name >> type;

            if (registered_numbers.count(number)) {
                std::cout << "NOMOR SUDAH ADA\n";
            } else if (emergency_q.size() + regular_q.size() >= MAX_PATIENTS) {
                 std::cout << "ANTRIAN PENUH\n";
            } else {
                registered_numbers.insert(number);
                Patient new_patient = {number, name};
                if (type == "E") {
                    emergency_q.push(new_patient);
                } else {
                    regular_q.push(new_patient);
                }
                std::cout << "TERDAFTAR\n";
            }
        } else if (command == "PANGGIL") {
            if (!emergency_q.empty()) {
                Patient p = emergency_q.front();
                emergency_q.pop();
                std::cout << "PANGGIL " << p.number << " " << p.name << "\n";
            } else if (!regular_q.empty()) {
                Patient p = regular_q.front();
                regular_q.pop();
                std::cout << "PANGGIL " << p.number << " " << p.name << "\n";
            } else {
                std::cout << "ANTRIAN KOSONG\n";
            }
        } else if (command == "LIHAT") {
            if (emergency_q.empty() && regular_q.empty()) {
                std::cout << "ANTRIAN KOSONG\n";
            } else {
                std::cout << "EMERGENCY: ";
                print_queue(emergency_q);
                std::cout << "\n";
                std::cout << "REGULAR: ";
                print_queue(regular_q);
                std::cout << "\n";
            }
        } else if (command == "CARI") {
            int number_to_find;
            std::cin >> number_to_find;
            bool found = false;

            std::queue<Patient> temp_e = emergency_q;
            int pos = 1;
            while (!temp_e.empty()) {
                if (temp_e.front().number == number_to_find) {
                    std::cout << "POSISI " << pos << " EMERGENCY\n";
                    found = true;
                    break;
                }
                temp_e.pop();
                pos++;
            }

            if (!found) {
                std::queue<Patient> temp_r = regular_q;
                pos = 1;
                while (!temp_r.empty()) {
                    if (temp_r.front().number == number_to_find) {
                        std::cout << "POSISI " << pos << " REGULAR\n";
                        found = true;
                        break;
                    }
                    temp_r.pop();
                    pos++;
                }
            }

            if (!found) {
                std::cout << "TIDAK DITEMUKAN\n";
            }
        }
    }

    return 0;
}
